### ISSUE TITLE
`regexp/negation` false positives because of `i` flag

### ISSUE THREAD

**Issue by RunDevelopment** [ota-meshi/eslint-plugin-regexp#266]
See the following example:

```js
var re2 = /[^\P{Ll}]/iu;
```

![image](https://user-images.githubusercontent.com/20878432/124463287-f9bfea80-dd92-11eb-9e42-0c3524d89b30.png)

Ignoring the strange suggested fix (this will be a separate issue), the report is incorrect. 

`/[^\P{Ll}]/iu` != `/\p{Ll}/iu`: 
The Canonicalize operation of the `i` flag is applied _before_ the character class is inverted. This example, `\P{Ll}` means "all character that are not lower-case letters". Since`\p{Ll}` contains all upper-case letters, it will also contain all lower-case letters after the Canonicalize operation (not quite right but almost), so `\P{Ll}` matches (almost) all characters. The character class will then match (almost) no character because it is inverted. This is obviously very different from `/\p{Ll}/iu` which matches (almost) all letter characters.

**Comment by RunDevelopment** [comment]
This is a bug in the `regexp/negation` rule but we can't fix it right now because of an upstream bug in refa. The upstream bug has already been fixed and I'll try to release a new refa version by end of the week.

**Comment by RunDevelopment** [comment]
Oops, I accidentally tagged this issue for closing in #273.
